src/ui/components/Library/FilterBar.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ChangeEvent, KeyboardEvent, useContext } from "react";
+
+import { TextInput } from "../shared/Forms";
+
+import { FilterDropdown } from "./FilterDropdown";
+import { LibraryFilters, LibraryFiltersContext } from "./useFilters";
+
+export function FilterBar({
+  displayedString,
+  setDisplayedText,
+  setAppliedText,
+}: {
+  displayedString: string;
+  setDisplayedText: (str: string) => void;
+  setAppliedText: (str: string) => void;
+}) {
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setDisplayedText(e.target.value);
+  };
+  const onKeyPress = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
```

Comment:
What should we do on _blur_ (or "Escape" key)?

Currently the pending filter text _won't_ be applied, which is good, but there's no visual indicator that the filter text is out of sync with the filters you're viewing.

Seems like we should discard the pending text on blur/escape?

Reply:
I was looking for some examples of how this is handled and it looks like the answer is, it isn't? Specifically in the GitHub filter example, pressing escape while focused is a noop, and if you do blur by clicking elsewhere, the pending change isn't discarded. Not a big fan for the same reasons you indicated. But at the same time, I don't think I'd expect for the text to be discarded on blur.

I think the main problem is that we don't indicate to the user what filter text is currently applied. If that were somewhere, then at least you can somewhat infer whether your change has been applied or not.

Maybe we should show something like, "Showing 89 replays that match <insert-filter-text-here>"? That gets wordy though.

Reply:
Ah, good point. GitHub is an interesting example.

I agree with you that blowing it away on blur might not be great, but I think it's not a given that people will know to press Enter to confirm. Could we just apply on blur? (And maybe discard on Escape?)

Alternately maybe we could use color to _hint_ at an unsaved/unapplied change?

Reply:
It's probably not optimal but I think I'm satisfied with being at parity here (for now) and then we can readjust if it's a big problem. Let me know what you think!

Reply:
Not worth holding up the merge for this, but I'd prefer we follow up with a fix to more gracefully handle that case 😁  I don't mind doing it.

Reply:
Down, thanks! I'll try to race you to it too